pygame/math.py

```diff
@@ -0,0 +1,310 @@
+                        self.y = float(tokens[1])
+                    except ValueError:
+                        raise TypeError("Invalid string arguments - not floats ({}).".
+                                        format(args[0]))
+                else:
+                    raise TypeError("Invalid string argument - not like __repr__ ({}).".
+                                    format(args[0]))
+            elif len(args[0]) == 2:
+                self.x = args[0][0]
+                self.y = args[0][1]
+            else:
+                raise TypeError("Invalid argument length ({}).".format(len(args[0])))
+        else:
+            self.x = args[0]
+            self.y = args[1]
+
+    def __repr__(self):
+        return "<Vector2({}, {})>".format(self.x, self.y)
+
+    def __str__(self):
+        return "[{}, {}]".format(self.x, self.y)
+
+    def __len__(self):
+        return 2
+
+    def __getitem__(self, key):
+        """Provide indexed read access (vec[0] is x vec[1] is y)."""
+        if isinstance(key, slice):
+            return [self[ind] for ind in xrange(*key.indices(len(self)))]
+        elif isinstance(key, int):
+            if key < 0:
+                key += len(self)
+            if key == 0:
+                return self.x
+            elif key == 1:
+                return self.y
+            else:
+                raise IndexError("Out of range index requested: {}".format(key))
+        else:
+            raise TypeError("Invalid argument type")
+
+    def __setitem__(self, key, value):
+        """Provide indexed modification."""
+        if isinstance(value, Vector2):
+            value = [value.x, value.y]
+        if isinstance(key, slice):
+            indices = range(*key.indices(len(self)))
+            if len(value) <= len(self) and len(value) == len(indices):
+                for count, index in enumerate(indices):
+                    self[index] = value[count]
+            else:
+                raise ValueError("Invalid slice or value arguments ({}).".format(key))
+        elif isinstance(key, int):
+            if key < 0:
+                key += len(self)
+            if key == 0:
+                self.x = value
+            elif key == 1:
+                self.y = value
+            else:
+                raise IndexError("Out of range index requested: {}".format(key))
+        else:
+            raise TypeError("Invalid argument type")
+
+    def __delitem__(self):
+        """Item deletion not supported."""
+        raise TypeError("Items may not be deleted.")
+
+    def __eq__(self, other):
+        if isinstance(other, Vector2):
+            return (self.x == other.x) and (self.y == other.y)
+        elif hasattr(other, '__iter__'):
+            try:
+                other_v = Vector2(other)
+                return self == other_v
+            except TypeError:
+                # Doesn't seem to be vector2-like, so NotImplemented
+                return False
+        return False
```

Comment:
I think this should `return NotImplemented` instead of `return False` in both cases (see https://docs.python.org/2/library/stdtypes.html#the-notimplemented-object).